cat-launcher/src/pages/PlayPage/index.tsx

```diff
@@ -1,18 +1,75 @@
-import { useQuery } from "@tanstack/react-query";
+import {
+    DndContext,
+    closestCenter,
+    KeyboardSensor,
+    PointerSensor,
+    useSensor,
+    useSensors,
+    DragEndEvent,
+} from "@dnd-kit/core";
+import {
+    arrayMove,
+    SortableContext,
+    sortableKeyboardCoordinates,
+    verticalListSortingStrategy,
+} from "@dnd-kit/sortable";
+import { useMutation, useQuery, useQueryClient } from "@tanstack/react-query";
+import { useEffect, useState } from "react";
 
 import GameVariantCard from "./GameVariantCard";
-import { fetchGameVariantsInfo } from "@/lib/commands";
+import {
+    fetchGameVariantsInfo,
+    updateGameVariantOrder,
+} from "@/lib/commands";
+import { GameVariantInfo } from "@/generated-types/GameVariantInfo";
 
 function PlayPage() {
+    const queryClient = useQueryClient();
     const {
         data: gameVariantsInfo = [],
         isLoading: gameVariantsLoading,
         isError: gameVariantsError,
         error: gameVariantsErrorObj,
-    } = useQuery({
+    } = useQuery<GameVariantInfo[]>({
         queryKey: ["gameVariantsInfo"],
         queryFn: fetchGameVariantsInfo,
     });
+    const [orderedItems, setOrderedItems] = useState<GameVariantInfo[]>([]);
+
+    useEffect(() => {
+        if (gameVariantsInfo) {
+            setOrderedItems(gameVariantsInfo);
+        }
+    }, [gameVariantsInfo]);
+
+    const sensors = useSensors(
+        useSensor(PointerSensor),
+        useSensor(KeyboardSensor, {
+            coordinateGetter: sortableKeyboardCoordinates,
+        }),
+    );
+
+    const { mutate } = useMutation({
```

Comment:
Show error to the user if the mutate fails. Also, rename mutate to something more descriptive.